### ISSUE TITLE
AutoSerializable source generator does not work for .NET 5/6 business libraries

### ISSUE THREAD

**Issue by TheCakeMonster** [MarimerLLC/csla#2659]
The new source generator for POCO auto-serialization built in issue #2148 does not work in a business library targeting anything other than .NET Standard.

Warnings are generated saying that Csla.dll cannot be found. Here is an example:

Warning CS8034 - Unable to load Analyzer assembly C:\Users\Andrew\source\repos\MarimerLLC\csla\Bin\Debug\netstandard2.0\Csla.dll: Could not find file 'C:\Users\Andrew\source\repos\MarimerLLC\csla\Bin\Debug\netstandard2.0\Csla.dll'

CSLA version: 6 (preview)
OS: Windows

I think the problem is that we are referencing Csla.dll. For a .NET 5/6 library, the platform-specific version of Csla will be used. However, only .NET Standard 2.0 libraries can be referenced by analyzers and source generators, as Roslyn is not able to load .NET Core assemblies.

**Comment by TheCakeMonster** [comment]
I intend to resolve this today. I think removing the reference to Csla.dll from Csla.Generators.CSharp will fix the problem.

**Comment by TheCakeMonster** [comment]
Hmm, bad time to be trying to fix issues with source generators, it seems.

Debugging/development experience on source generators appears to be almost completely broken in Roslyn. Sometimes, some versions of VS might work. Vaguely. Occasionally.

https://github.com/dotnet/roslyn/issues/49249

6 hours' work, and not a single line of code debugged, in any of 3 versions of Visual Studio.

_Sigh_

**Comment by TheCakeMonster** [comment]
Nope, source generators don't even run at the moment. I have confirmed this with my original POC from my own repo, which doesn't suffer from the 100% CPU utilisation problem I have been battling most of the day today.

It appears that this bit of .NET has been broken somehow. I'll have to park this and come back to it once it has been fixed - although I don't know when this is likely to get fixed. It's surprising to me that .NET 6 got released without this working.

There is a new, preferable source generation method in .NET 6 which is supposed to have less of a performance impact. At some point I will have to look at upgrading to that - but it feels too soon to do that at the moment, because .NET 6 probably cannot be considered ubiquitous yet.

**Comment by TheCakeMonster** [comment]
I've pushed the changes that I think might work to the new SourceGen_Dereference branch in my repo, but I have no way of testing whether they work, as source generators aren't running at all for me at the moment, let alone being debuggable.

**Comment by rockfordlhotka** [comment]
I did remove the generators from the nuspec file (`/nuget/definitions/csla.nuspec`), so maybe that's why they aren't running for you?

I had to remove them because they had that circular reference to Csla.dll that was causing nuget issues as well as the issues you talk about in this thread.

**Comment by TheCakeMonster** [comment]
Thanks Rocky. I did wonder if that might be the case, as I'd not seen the warnings when I'd looked. However, I'm pretty confident it's not anything we've done.

I did mis-speak, in that it's not that they aren't running, but that the code change I have tried to make doesn't work, so it _appears_ that they aren't running. My bad.

The problem is that we seem to have been left with no way to debug them, so I can't work out _why_ the changes don't work. We used to be able to attach a debugger, but that appears to have been disabled, or broken, or something. Outputting data using Console.WriteLine and Debug.WriteLine doesn't emit anything anywhere I can find. I'm running out of ideas - and time.

Debugging source generators has been a bit imperfect for a while, and we were prepared to wait - after all, agile is about experimentation. However, today is the first time in a long time when I feel truly let down by the .NET team. It's been about 18 months since source generators was introduced, and yet we are in the worst position possible now - no way at all that I can think of to debug them - not even 1980s-style print statements.

After 18 months, I think it's fairly awful that we're in this state. At some point the experiment should end.

I did think of applying to Microsoft for a job, but today I am very glad I decided not to. Today I find myself very, very pleased to be able to say I am not, in any way, associated with the addition of source generators to .NET.
